src/lib/route-builder.js

```diff
@@ -114,45 +108,67 @@ class RouteBuilder {
    */
   * getDestinationTransfer (sourceTransfer) {
     log.info('constructing destination transfer ' +
-      'sourceLedger=%s sourceAmount=%s ilpHeader=%s',
-      sourceTransfer.ledger, sourceTransfer.amount,
-      sourceTransfer.data && JSON.stringify(sourceTransfer.data.ilp_header))
-    const ilpHeader = sourceTransfer.data && sourceTransfer.data.ilp_header
-    if (!ilpHeader) {
-      throw new Error('source transfer is missing ilp_header in memo')
+      'sourceLedger=%s sourceAmount=%s ilpPacket=%s',
+      sourceTransfer.ledger, sourceTransfer.amount, sourceTransfer.ilp)
+    if (!sourceTransfer.ilp) {
+      throw new IlpError({
+        code: 'S01',
+        name: 'Invalid Packet',
+        message: 'source transfer is missing "ilp"'
+      })
     }
+    let ilpPacket
+    try {
+      ilpPacket = packet.deserializeIlpPayment(Buffer.from(sourceTransfer.ilp, 'base64'))
+    } catch (err) {
+      log.debug('error parsing ILP packet: ' + sourceTransfer.ilp)
+      throw new IlpError({
+        code: 'S01',
+        name: 'Invalid Packet',
+        message: 'source transfer has invalid ILP packet'
+      })
+    }
+    log.debug('constructing destination transfer ' +
```

Comment:
This is a little deceptive because that's the final destination, not the "destination transfer" as far as the connector is concerned